packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -241,10 +260,15 @@ class ScrollDragController implements Drag {
   @override
   void update(DragUpdateDetails details) {
     assert(details.primaryDelta != null);
-    _lastDetails = details;
     double offset = details.primaryDelta;
-    if (offset == 0.0)
+    if (offset == 0.0) {
+      // No axis motion over 2 consecutive frames breaks previous momentum.
+      if (_lastDetails is DragUpdateDetails && _lastDetails.primaryDelta == 0.0) {
```

Comment:
This will never be true, since you never set _lastDetails when its primaryDelta would have been 0.0.